### PR TITLE
feat(common/lmlayer): allow for verbose word breaker specification

### PR DESCRIPTION
**Pull request by eddieantonio**
This introduces an expanded way that the word breaker can be specified.

Whereas before, we'd have this:

```typescript
{
    wordBreaker: 'default'
}
```

Now we _may_ have this:

```typescript
{
    wordBreaker: {
        use: 'default',
    }
}
```

This makes it easier to add options to the word breaker specification (see #2753), as the options can be provided as additional properties to the `wordBreaker` object. e.g., in the future, we could have this:

```typescript
{
    wordBreaker: {
        use: 'default',
        joinWordsAt: ['-'],
    }
}
```

This change is backward-compatible. The old way of specifying word breakers still works.

### CODE REVIEW

File: developer/js/source/lexical-model-compiler/lexical-model-compiler.ts [hunk 1]
```diff
@@ -97,20 +92,36 @@ export class ModelSourceError extends Error {
...
       .replace(/^wordBreak(ing|er)\b/, 'function');
   }
 }
+
+/**
+ * Given a word breaker specification in any of the messy ways,
+ * normalizes it to a common form that the compiler can deal with.
+ */
+function normalizeWordBreakerSpec(wordBreakerSpec: LexicalModelSource["wordBreaker"]): WordBreakerSpec {
+  if (wordBreakerSpec == undefined) {
```
Comment (mcdurdin) [inline developer/js/source/lexical-model-compiler/lexical-model-compiler.ts:112]: Previously you tested `!wordBreakerSpec`. Is that identical to `wordBreakerSpec == undefined`?

File: developer/js/source/lexical-model-compiler/lexical-model.ts [hunk 2]
```diff
@@ -9,7 +9,20 @@ interface LexicalModelDeclaration {
   //... metadata ...
 }
 
-type WordBreakerSpec = 'default' | 'ascii' | WordBreakingFunction;
+/**
+ * Keyman 13.0+ word breaker specification:
```
Comment (mcdurdin) [inline developer/js/source/lexical-model-compiler/lexical-model.ts:13]:
```suggestion
 * Keyman 14.0+ word breaker specification:
```

File: developer/js/tests/fixtures/example.qaa.wordbreaker/example.qaa.wordbreaker.model.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const source: LexicalModelSource = {
+  format: 'trie-1.0',
+  sources: ['wordlist.tsv'],
+  /* Keyman 13.0+ word breaker specification: */
```
Comment (mcdurdin) [inline developer/js/tests/fixtures/example.qaa.wordbreaker/example.qaa.wordbreaker.model.ts:4]:
```suggestion
  /* Keyman 14.0+ word breaker specification: */
```

File: developer/js/source/lexical-model-compiler/lexical-model-compiler.ts (hunk 1 above)
Comment (eddieantonio) [reply developer/js/source/lexical-model-compiler/lexical-model-compiler.ts:112]: `wordBreaker == undefined` is true when `wordBreaker === null` or `wordBreaker === undefined`, and is false otherwise. `!wordBreaker` is a bit more permissive in that it allows for `wordBreaker === 0` and `wordBreaker === ''`.

Comment (eddieantonio) [reply developer/js/source/lexical-model-compiler/lexical-model-compiler.ts:112]: I got bored so I decided to make a truth table:

`v`         | `v == undefined` | `!v`   | Desired
------------|------------------|--------|------------
`undefined` | `true`           | `true` | `true`
`null`      | `true`           | `true` | `true`
`0`         | `false`          | `true` | `false`
`false`     | `false`          | `true` | `false`
`''`        | `false`          | `true` | `false`
`NaN`       | `false`          | `true` | `false`

Basically, `v == undefined` gives me the behaviour I want: only `true` when wordBreaker is undefined or null.
